src/d3a/d3a_core/sim_results/area_statistics.py

```diff
@@ -592,3 +591,41 @@ def _convert_output_format(self, price_energy, csv_output, redis_output):
                 } for timeslot, trades in trade_rates.items()
             ]
             redis_output[node.uuid] = deepcopy(csv_output[node.name])
+
+
+class BaselinePeakEnergyStats:
+    def __init__(self):
+        self.baseline_peak_percentage_result = {}
+        self.baseline_peak_percentage_result_redis = {}
+
+    def update(self, area):
+        self.update_exported_imported_energy(area)
+        self.results_to_redis(area)
+
+    def update_exported_imported_energy(self, area):
+        if area.baseline_energy_settings is not None:
+            area.stats.aggregate_exported_imported_energy(area)
+            baseline_import = area.baseline_energy_settings["baseline_peak_energy_import_kWh"]
+            baseline_export = area.baseline_energy_settings["baseline_peak_energy_export_kWh"]
+            if baseline_import > 0:
+                create_subdict_or_update(self.baseline_peak_percentage_result, area.name,
+                                         {"import": {k: v/baseline_import
+                                          for k, v in area.stats.imported_energy.items()}})
+            if baseline_export > 0:
+                create_subdict_or_update(self.baseline_peak_percentage_result, area.name,
```

Comment:
I would suggest to refactor this into a small function that updates both `self.baseline_peak_percentage_result` and `self.baseline_peak_percentage_result_redis` in one go, in order to avoid the second recursion. You can use the area.uuid parameter, in order to avoid the area name-> uuid conversion.

Reply:
yes, good Idea! will do!